UpdatedSyntheticDataset/SyntheticDataset2/ElementsCreator/background.py

```diff
@@ -1,36 +1,43 @@
 from PIL import Image
-import random
-import os
+import random, os
 
 class BackgroundGenerator(object):
 
-    def __init__(self,width,height,path_to_backrounds):
+    def __init__(self,path_to_backrounds):
         """
-        Return a background for a Target
+        Create a background for a Target or TargetField
 
-        :param width: width of background
-        :type width: int
-        :param height: height of background
-        :type height: int
-        :param path_to_backrounds: location of backgrounds directory on computer
+        :param path_to_backrounds: location of backgrounds directory
         """
-        self.width=width
-        self.height=height
         self.path_to_backrounds=path_to_backrounds
 
-    def generate_background(self):
-        try:
-            os.remove(self.path_to_backrounds + "/.DS_Store")
-        except:
-            pass
-        background = Image.open(self.path_to_backrounds + "/" + random.choice(os.listdir(self.path_to_backrounds)))
-        crop_coordinates = self.get_random_coordinates(background.width,background.height)
+    def generate_full_background(self):
+        """
+        Return a full sized background
+        """
+        image_list = os.listdir(self.path_to_backrounds)
+        image_list = [f for f in image_list if not f.startswith(".")]
+
+        background = Image.open(self.path_to_backrounds + "/" + random.choice(image_list))
+        return background
+
+    def generate_specific_background(self,specified_width,specified_height):
+        """
+        Return a background with a specified size
+
+        :param width: width of background in pixels
+        :type width: int
+        :param height: height of background in pixels
+        :type height: int
+        """
+        background = self.generate_full_background()
+        crop_coordinates = self.get_random_coordinates(background.width,background.height,specified_width,specified_height)
         background = background.crop(crop_coordinates)
         return background
 
-    def get_random_coordinates(self,background_width,background_height):
-        x1 = random.randint(0,background_width-self.width)
-        y1 = random.randint(0,background_height-self.height)
-        x2 = x1+self.width
-        y2 = y1+self.height
+    def get_random_coordinates(self,background_width,background_height,specified_width,specified_height):
```

Comment:
background_width, background_height should be changed to something like start_x, start_y